examples/webgl2_texture2darray_compressed.html

```diff
@@ -0,0 +1,313 @@
+
+		<script type="importmap">
+			{
+				"imports": {
+					"three": "../build/three.module.js",
+					"three/addons/": "./jsm/"
+				}
+			}
+		</script>
+
+		<script type="module">
+
+			import * as THREE from 'three';
+
+			import { GLTFLoader } from 'three/addons/loaders/GLTFLoader.js';
+			import { OrbitControls } from 'three/addons/controls/OrbitControls.js';
+			import { KTX2Loader } from 'three/addons/loaders/KTX2Loader.js';
+			import { RoomEnvironment } from 'three/addons/environments/RoomEnvironment.js';
+
+			let container, clock, mixer, actions, activeAction;
+			let camera, scene, renderer, model;
+
+			init();
+			animate();
+
+			function init() {
+
+				container = document.createElement( 'div' );
+				document.body.appendChild( container );
+
+				camera = new THREE.PerspectiveCamera( 45, window.innerWidth / window.innerHeight, 0.25, 100 );
+				camera.position.set( 1, 1, 5 );
+
+				scene = new THREE.Scene();
+
+				clock = new THREE.Clock();
+
+
+				const dirLight = new THREE.DirectionalLight( 0xffffff, 1 );
+				dirLight.position.set( 5, 5, 5 );
+				scene.add( dirLight );
+
+				// ground
+				const mesh = new THREE.Mesh( new THREE.PlaneGeometry( 2000, 2000 ), new THREE.MeshPhongMaterial( { color: 0xf1f1f1, depthWrite: false } ) );
+				mesh.rotation.x = - Math.PI / 2;
+				scene.add( mesh );
+
+				const grid = new THREE.GridHelper( 200, 40, 0x000000, 0x000000 );
+				grid.material.opacity = 0.2;
+				grid.material.transparent = true;
+				scene.add( grid );
+
+
+				renderer = new THREE.WebGLRenderer( { antialias: true } );
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				renderer.outputEncoding = THREE.sRGBEncoding;
+				container.appendChild( renderer.domElement );
+
+
+				const environment = new RoomEnvironment();
+				const pmremGenerator = new THREE.PMREMGenerator( renderer );
+
+				scene.background = new THREE.Color( 0xbbbbbb );
+				scene.environment = pmremGenerator.fromScene( environment ).texture;
+				environment.dispose();
+
+				const ktx2Loader = new KTX2Loader();
+				ktx2Loader.setTranscoderPath( 'js/libs/basis/' );
+				ktx2Loader.detectSupport( renderer );
+
+				// model
+
+				function debug( scene, material ) {
+
+					const head_debug = new THREE.Mesh( new THREE.PlaneGeometry(), material );
+					head_debug.position.set( 1.25, 1.6, 0 );
+					head_debug.geometry.setAttribute( 'textureId', new THREE.BufferAttribute( new Int8Array( head_debug.geometry.index.count ).fill( 1 ), 1 ) );
+
+					const body_debug = new THREE.Mesh( new THREE.PlaneGeometry(), material );
+					body_debug.position.set( 1.25, 0.5, 0 );
+
+					body_debug.geometry.setAttribute( 'textureId', new THREE.BufferAttribute( new Int8Array( head_debug.geometry.index.count ).fill( 0 ), 1 ) );
+
+					return new THREE.Group().add( head_debug, body_debug );
+
+				}
+
+				function obc( material ) {
+
+					material.onBeforeCompile = function ( shader ) {
```

Comment:
Sorry for the late feedback but I have encountered this section just now.

Would it be possible to showcase compressed array textures differently and not by patching built-in materials? In the past, we made bad experiences with such examples since they are hard to maintain (they easily break when we refactor shaders). Could you use a custom shader material with minimal code (similar to https://threejs.org/examples/webgl2_buffergeometry_attributes_integer)?

Reply:
Simplest option here might be to copy https://threejs.org/examples/?q=array#webgl2_materials_texture2darray using THREE.CompressedArrayTexture? That example probably makes it more clear how the array texture is being used. 

Not opposed to doing something different here, but also don't want to hold the PR back.